Source/WebKit/Shared/haiku/AuxiliaryProcessMainHaiku.h

```diff
@@ -28,33 +29,114 @@
 #include "AuxiliaryProcess.h"
 #include "WebKit2Initialize.h"
 #include <wtf/RunLoop.h>
+#include <Application.h>
+#include <Looper.h>
+#include <MessageQueue.h>
+#include <String.h>
+#include <map>
+
+using namespace std;
 
 namespace WebKit {
 
-struct AuxiliaryProcessSettings 
-{
+class AuxiliaryProcessMainBase {
+public:
+    virtual bool platformInitialize(char* sig) { return true; }
+    virtual bool parseCommandLine(int argc, char** argv);
+    virtual void platformFinalize() { }
+	
     AuxiliaryProcessInitializationParameters&& takeInitializationParameters() { return WTFMove(m_parameters); }
+
+protected:
     AuxiliaryProcessInitializationParameters m_parameters;
 };
 
+class ProcessApp : public BApplication
+{
+	public:
+	BLooper* messageForward ;
+	status_t result;
+	map<string,BLooper*> proxy;
+	BMessageQueue stash;
+	ProcessApp(char* signature):BApplication(signature)
+	{
+		messageForward = nullptr;
+	}
+	void ProcessMessage(BMessage* message)
+	{
+		const char* tempStr;
+		BLooper* tempLooper;
+		message->FindString("identifier",&tempStr);
+		message->FindPointer("looper",(void**)&tempLooper);
+		string temp(tempStr);
+		proxy[temp] = tempLooper;
+		while(!stash.IsEmpty())
+		{
+			tempLooper->PostMessage(stash.NextMessage(),tempLooper->PreferredHandler());
+		}
+	}
+	void AttachAndSend(BMessage* message)
+	{
+		const char* tempStr;
+		message->FindString("identifier",&tempStr);
+		string temp(tempStr);
+		BLooper *looper = proxy[temp];
+
+		message = DetachCurrentMessage();
+		if(!looper)
+		{
+			stash.AddMessage(message);
```

Comment:
can you explain this? If the message is sent to an invalid target, can't we just ignore it?

Reply:
I added this because if the intended process is not launched yet but other process starts sending messages we need to store them and once they are ready to go we pop one by one

Reply:
Does this happen? And is it a problem if the messages get lost?

I think the process should be something like:
- WebProcess is started
- WebProcess sends a message to UI to tell its alive
- UIProcess does not send anything until WebProcess is ready

So we should not get messages before everything is ready?

Reply:
Yes the web process starts and then only ui process sends but in our case after we process starts only our application comes to life so we should not miss the first few messages

Reply:
So we have an initialization order problem. Maybe the UI process needs to send a "I'm ready" message to the web process when it is ?

Reply:
No I guess this is not a problem because when our process is ready to receive then we stash them actually ui process does the same within it stashes until the process is running when we say did finish launching then the messages are sent to processes until then it is stashed